ui/js/component/common.js

```diff
@@ -42,6 +42,59 @@ export class TruncatedText extends React.PureComponent {
   }
 }
 
+export class TruncatedMarkdown extends React.PureComponent {
+  static propTypes = {
+    lines: React.PropTypes.number,
+  };
+
+  static defaultProps = {
+    lines: null,
+  };
+
+  truncateMarkdown(text) {
```

Comment:
This entire function terrifies me. I'm guessing the proper way to do this is to fully parse the markdown, then return the plain text.